BrainPortal/public/javascripts/cbrain.js

```diff
@@ -790,6 +790,56 @@
       tool_tip.hide();
     });
 
+
+    // Adds copy to buffer button to all the element of class 'copy-text'.
+
+    if (navigator.clipboard) {
+      $(".copy-text").each(function () {
+        let $this = $(this);
+
+        // Keep line breaks
+        let text = $this.html()
+            .replace(/<br\s*\/?>/g, "\n")
+            .replace(/<\/p>\s*<p>/g, "\n")
+            .replace(/<\/div>\s*<div>/g, "\n")
+            .replace(/<\/?[^>]+(>|$)/g, "")
+            .trim();
+
+        if ($this.find(".copy-button").length === 0) {
+          let copyBtn = $(`
```

Comment:
I don't understand why you add the button and SVG image in the javasript code. Shouldn't this be just in the HTML template already? The JS code should only take care of the event triggered by a click, and make the copy to the clipboard. Everything visible in the page should already be in the view template, one way or another.

Reply:
But if you want to make it reusable, create a helper method! That's what helpers are for. The folder BrainPortal/app/helpers is full of them. Right now if someone wants to change the layout of the copy-to-clipboard button, we have to go into the javascript, that doesn't make any sense. The javascript code should do only the binding and event processing for copying to the clipboard, all the button and layouts should be performed by a helper. Something like:

```erb
Your token is <%= @token %> <%= copy_to_clipboard_button @token %>
```

which would generate the HTML, and put the token inside an attribute "data-clipboard" within the button. No need to fetch the text anywhere else, or parse it. Just a button.

Reply:
ok